src/python/WMCore/MicroService/MSPileup/MSPileupTaskManager.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+"""
+File       : MSPileupTaskManager.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: MSPileupTaskManager handles MSPileupTasks
+
+In particular, it perform the following tasks each polling cycle:
+    - fetches pileup sizes for all pileup documents in database back-end
+    - update RSE quotas
+    - perform monitoring task
+    - perform task for active pileups using up-to-date RSE quotas
+    - perform task for inactive pileups
+"""
+
+# system modules
+import os
+from threading import current_thread
+
+# WMCore modules
+from WMCore.MicroService.MSCore.MSCore import MSCore
+from WMCore.MicroService.DataStructs.DefaultStructs import PILEUP_REPORT
+from WMCore.MicroService.MSPileup.MSPileupData import MSPileupData
+from WMCore.MicroService.MSPileup.MSPileupTasks import MSPileupTasks
+from WMCore.MicroService.MSTransferor.DataStructs.RSEQuotas import RSEQuotas
+from WMCore.MicroService.Tools.PycurlRucio import getPileupContainerSizesRucio, getRucioToken
+from WMCore.Services.Rucio.Rucio import Rucio
+
+
+class MSPileupTaskManager(MSCore):
+    """
+    MSPileupTaskManager handles MSPileup tasks
+    """
+
+    def __init__(self, msConfig, **kwargs):
+        super().__init__(msConfig, **kwargs)
+        self.marginSpace = msConfig.get('marginSpace', 1024**4)
+        self.rucioAccount = msConfig.get('rucioAccount', 'ms-pileup')
+        self.rucioUrl = msConfig.get('rucioHost', 'http://cms-rucio.cern.ch')
+        self.rucioAuthUrl = msConfig.get('authHost', 'https://cms-rucio-auth.cern.ch')
+        creds = {"client_cert": os.getenv("X509_USER_CERT", "Unknown"),
+                 "client_key": os.getenv("X509_USER_KEY", "Unknown")}
+        configDict = {'rucio_host': self.rucioUrl, 'auth_host': self.rucioAuthUrl,
+                      'creds': creds, 'auth_type': 'x509'}
+        self.rucioClient = Rucio(self.rucioAccount, configDict=configDict)
+        self.dataManager = MSPileupData(msConfig)
+        self.mgr = MSPileupTasks(self.dataManager, self.logger,
+                                 self.rucioAccount, self.rucioClient)
+        self.rseQuotas = RSEQuotas(self.rucioAccount, msConfig["quotaUsage"],
+                                   minimumThreshold=msConfig["minimumThreshold"],
+                                   verbose=msConfig['verbose'], logger=self.logger)
+
+    def status(self):
```

Comment:
Along what we have been discussing, if we decide to split MSPileup between:
* MSPileup serving HTTP requests
* MSPileup executing the background tasks

how would we make the task-related information accessible to the HTTP-based service? The only way I see is by persisting that data in the database, right?